Formula/s/semgrep.rb

```diff
@@ -335,7 +336,9 @@ def install
 
     venv.pip_install_and_link buildpath/"cli"
 
-    generate_completions_from_executable(bin/"semgrep", shells: [:fish, :zsh], shell_parameter_format: :click)
+    generate_completions_from_executable(bin/"semgrep", "--legacy",
+                                         shells:                 [:fish, :zsh],
+                                         shell_parameter_format: :click)
```

Comment:
Can we update this to use the new implementation? `--legacy` makes it seem like it'll break at some point.

Reply:
I don't think it would be easy. The new implementation is wholly written in OCaml (versus a mix of OCaml and Python) and uses [cmdliner](https://github.com/dbuenzli/cmdliner) instead of click, which doesn't have support for shell completions currently (https://github.com/dbuenzli/cmdliner/issues/1).

I'm happy to update this again once we have shell completions available in the newer implementation.

For some additional colour, the newer implementation is incomplete and falls back to the old one for a subset of options/subcommands. So there is more feature work which needs to be done before removing the old one too. This flag is just to force that fallback to unconditionally occur since click knows to generate completions based off of environment variables, but we wouldn't normally fall back to the click based implementation when no arguments are present on the command line.